components/script/stylesheet_loader.rs

```diff
@@ -98,82 +97,29 @@ pub struct StylesheetContext {
 
 impl StylesheetContext {
     fn unminify_css(&self, data: Vec<u8>, file_url: ServoUrl) -> Vec<u8> {
-        if !self.document.root().window().unminify_css() {
+        if self.document.root().window().unminified_css_dir().is_none() {
             return data;
         }
 
         let mut style_content = data;
 
-        // Write the minified code to a temporary file and pass its path as an argument
-        // to js-beautify --type css to read from. Meanwhile, redirect the process' stdout into
-        // another temporary file and read that into a string. This avoids some hangs
-        // observed on macOS when using direct input/output pipes with very large
-        // unminified content.
-        let (input, output) = (tempfile::NamedTempFile::new(), tempfile::tempfile());
-        if let (Ok(mut input), Ok(mut output)) = (input, output) {
-            input.write_all(&style_content).unwrap();
-            match Command::new("js-beautify")
-                .arg("--type")
-                .arg("css")
-                .arg(input.path())
-                .stdout(output.try_clone().unwrap())
-                .status()
-            {
-                Ok(status) if status.success() => {
-                    output.seek(std::io::SeekFrom::Start(0)).unwrap();
-                    output.read_to_end(&mut style_content).unwrap();
-                },
-                _ => {
-                    log::warn!(
-                        "Failed to execute js-beautify --type css. Will store unmodified css"
-                    );
-                },
+        if let Some((input, mut output)) = create_temp_files() {
+            if execute_js_beautify(
+                input.path(),
+                output.try_clone().unwrap(),
+                BeautifyFileType::Css,
```

Comment:
Is the BeautifyFileType and the directory the only pieces in the code that is different between scripts and stylesheets? It feels like we could still remove some more duplicated logic.

Reply:
`BeautifyFileType` is used only to add `--type css` to `js-beautify` and later `html`, but the main difference is how the content is read and written. For JS, there’s s`cript.code`, while for CSS, I’ve defined `style_content`

Reply:
@jdm what do you think? should I remove more duplicated logic?